components/director/internal/domain/api/service.go

```diff
@@ -173,6 +191,15 @@ func (s *service) RefetchAPISpec(ctx context.Context, id string) (*model.APISpec
 		return nil, err
 	}
 
+	fetchRequest, err := s.fetchRequestRepo.GetByReferenceObjectID(ctx, tnt, model.APIFetchRequestReference, id)
```

Comment:
When I'm trying to refetch api spec with mutation:
```gql
mutation refetch {
  refetchAPISpec(apiID: "b973beb9-1340-4724-afca-705cc3df0234") {
    data
    format
    type
    fetchRequest {
      url
      mode
      filter
      status {
        timestamp
        condition
      }
      auth {
        additionalHeaders
        credential {
          __typename
        }
      }
    }
  }
}
```

I'm getting following response with error:
```gql
{
  "errors": [
    {
      "message": "Internal Server Error: Cannot fetch FetchRequest. APIDefinition ID is empty",
      "path": [
        "refetchAPISpec",
        "fetchRequest"
      ]
    }
  ],
  "data": {
    "refetchAPISpec": {
      "data": "apiVersion: v1\ndescription: Compass\nname: compass\nversion: 0.1.0\n",
      "format": "YAML",
      "type": "OPEN_API",
      "fetchRequest": null
    }
  }
}
```